src/gui/qgsfilewidget.cpp

```diff
@@ -231,25 +231,40 @@ void QgsFileWidget::openFileDialog()
 
   // Handle Storage
   QString fileName;
+  QStringList fileNames;
   QString title;
   if ( mStorageMode == GetFile )
   {
     title = !mDialogTitle.isEmpty() ? mDialogTitle : tr( "Select a file" );
     fileName = QFileDialog::getOpenFileName( this, title, QFileInfo( oldPath ).absoluteFilePath(), mFilter );
   }
+  else if ( mStorageMode == GetMultipleFiles )
+  {
+    title = !mDialogTitle.isEmpty() ? mDialogTitle : tr( "Select one ore more files" );
+    fileNames = QFileDialog::getOpenFileNames( this, title, QFileInfo( oldPath ).absoluteFilePath(), mFilter );
+  }
   else if ( mStorageMode == GetDirectory )
   {
     title = !mDialogTitle.isEmpty() ? mDialogTitle : tr( "Select a directory" );
     fileName = QFileDialog::getExistingDirectory( this, title, QFileInfo( oldPath ).absoluteFilePath(),  QFileDialog::ShowDirsOnly );
   }
 
-  if ( fileName.isEmpty() )
+  if ( fileName.isEmpty() && fileNames.isEmpty( ) )
     return;
 
+  if ( mStorageMode != GetMultipleFiles )
+  {
+    fileName = QDir::toNativeSeparators( QDir::cleanPath( QFileInfo( fileName ).absoluteFilePath() ) );
+  }
+  else
+  {
+    for ( int i = 0; i < fileNames.length(); i++ )
+    {
+      fileNames.replace( i, QDir::cleanPath( QFileInfo( fileNames.at( i ) ).absoluteFilePath() ) ) ;
```

Comment:
Was the omission of toNativeSeparators deliberate here? It makes things much nicer for windows users.